crates/shared/src/sources/balancer_v2/pools/stable.rs

```diff
@@ -0,0 +1,65 @@
+//! Module implementing stable pool specific indexing logic.
+
+use super::{common, FactoryIndexing, PoolIndexing};
+use crate::sources::balancer_v2::graph_api::{PoolData, PoolType};
+use anyhow::{ensure, Result};
+use contracts::BalancerV2StablePoolFactory;
+
+#[derive(Clone, Debug, Default, Eq, PartialEq)]
+pub struct PoolInfo {
+    pub common: common::PoolInfo,
```

Comment:
If we compose here instead of aliasing is it really needed that `common::PoolInfo` is implementing `PoolIndexing` as well? There is quite a few structs with similar names

Reply:
No it isn't. Good point. I had originally aliased `stable::PoolInfo == common::PoolInfo` but moved away from it.

I will remove the `impl PoolIndexing for common::PoolInfo`.

Regarding multiple `PoolInfo` structs, this is just a question of personal preference. I tend to prefer writing:
```rust
use path::to::stable;
stable::PoolInfo

// --- OR ---

use path::to::stable::PoolInfo as StablePoolInfo;
StablePoolInfo
```

over
```rust
use path::to::stable::StablePoolInfo;
StablePoolInfo
```

Mostly because I find `stable::StablePoolInfo` looks a bit redundant. Happy to change this though, I don't have strong feelings about it.

Reply:
I like the first option of your preferences (before the OR).